llvm/lib/Transforms/Scalar/InferAddressSpaces.cpp

```diff
@@ -304,10 +304,24 @@ static bool isNoopPtrIntCastPair(const Operator *I2P, const DataLayout &DL,
          (P2IOp0AS == I2PAS || TTI->isNoopAddrSpaceCast(P2IOp0AS, I2PAS));
 }
 
+// Returns true if every user of a given GV are "simple" loads or stores in the
+// same function, with the store actually writing to GV.
+static bool isLocallyAccessedBySimpleLoadsStores(const GlobalVariable *GV,
+                                                 const Function *F) {
+  return all_of(GV->users(), [=](const User *U) {
+    if (const auto *SI = dyn_cast<StoreInst>(U))
+      return SI->getPointerOperand() == GV && SI->getFunction() == F;
```

Comment:
Can you add negative tests for the data use and the wrong function

Reply:
> Can you add negative tests for the data use and the wrong function

done